src/Engine/Scene/CameraComponent.cpp

```diff
@@ -0,0 +1,116 @@
+#include <Core/Asset/Camera.hpp>
+#include <Engine/Scene/CameraComponent.hpp>
+
+#include <Core/Containers/MakeShared.hpp>
+#include <Core/Math/Math.hpp>
+#include <Engine/Data/Mesh.hpp>
+#include <Engine/Data/PlainMaterial.hpp>
+#include <Engine/Data/ShaderConfigFactory.hpp>
+#include <Engine/Rendering/RenderObject.hpp>
+
+namespace Ra {
+
+using Core::Math::Pi;
+using Core::Math::PiDiv2;
+using Core::Math::PiDiv4;
+
+namespace Engine {
+
+namespace Scene {
+
+CameraComponent::CameraComponent( Entity* entity,
+                                  const std::string& name,
+                                  Scalar height,
+                                  Scalar width ) :
+    Component( name, entity ) {
+    m_camera = std::make_unique<Core::Asset::Camera>( height, width );
+}
+
+CameraComponent::~CameraComponent() = default;
+
+void CameraComponent::initialize() {
+    using Data::Material;
+    using Data::Mesh;
+    using Data::PlainMaterial;
+    if ( !m_renderObjects.empty() ) return;
+    // Create the render mesh for the camera
+
+    Ra::Core::Geometry::LineMesh triMesh;
+    triMesh.setVertices( {{0_ra, 0_ra, 0_ra},
+                          {-.5_ra, -.5_ra, -1_ra},
+                          {-.5_ra, .5_ra, -1_ra},
+                          {.5_ra, .5_ra, -1_ra},
+                          {.5_ra, -.5_ra, -1_ra},
+                          {-.3_ra, .5_ra, -1_ra},
+                          {0_ra, .7_ra, -1_ra},
+                          {.3_ra, .5_ra, -1_ra}} );
+    triMesh.setIndices(
+        {{0, 1}, {0, 2}, {0, 3}, {0, 4}, {1, 2}, {1, 4}, {2, 3}, {3, 4}, {5, 6}, {6, 7}} );
+
+    auto m = std::make_shared<Data::LineMesh>( m_name + "_mesh" );
+    m->loadGeometry( std::move( triMesh ) );
+    // Create the RO
+    auto mat              = Core::make_shared<PlainMaterial>( m_name + "_Material" );
+    mat->m_color          = {1_ra, .5_ra, 0_ra, 1_ra};
+    mat->m_perVertexColor = false;
+    Rendering::RenderTechnique rt;
+    auto cfg = Data::ShaderConfigurationFactory::getConfiguration( "Plain" );
+    rt.setConfiguration( *cfg );
+    rt.setParametersProvider( mat );
```

Comment:
When building a component, one must be agnostic to the renderer. RenderTechniques are specific to the renderer. Just define a Plain Material here and the camera should be visualized with all the renderer (or define it as a Debug or UI object). See GeometryComponent to build a RenderObject usable by any renderer.